src/sagemaker/session.py

```diff
@@ -826,6 +827,9 @@ def train(  # noqa: C901
         inferred_resource_config = update_nested_dictionary_with_values_from_config(
             resource_config, TRAINING_JOB_RESOURCE_CONFIG_PATH, sagemaker_session=self
         )
+        inferred_profiler_config = update_nested_dictionary_with_values_from_config(
+            profiler_config, TRAINING_JOB_PROFILE_CONFIG_PATH, sagemaker_session=self
+        )
```

Comment:
Same. Use `resolve_class_attribute_from_config` here as well.

Reply:
hmmm in estimator.py the profiler_config is of type ```ProfilerConfig``` , but in session.py the doc mentions it is just a dict ? Would we still need `resolve_class_attirbute_from_config` ?

Reply:
sycned offline - weird impl, can keep it as dict